src/SlimFaas/ReplicasService.cs

```diff
@@ -110,11 +99,11 @@ public Task CheckScaleAsync(string kubeNamespace)
         if (tasks.Count <= 0) return Task.CompletedTask;
 
         var updatedFunctions = new List<DeploymentInformation>();
-        
+
         foreach (var function in Deployments.Functions)
         {
-            var updatedFunction = tasks.FirstOrDefault(t => t.Result.Deployment == function.Deployment);
-            updatedFunctions.Add(function with { Replicas = updatedFunction != null ? updatedFunction.Result.Replicas : function.Replicas });
+            var updatedFunction = tasks.Find(t => t.Result?.Deployment == function.Deployment);
```

Comment:
In this foreach loop `.Result` Task property is accessed. The may cause deadlocks. We should have the following code before entering this loop.
```c#
ReplicaRequest?[] replicaResuests = await Task.WhenAll(tasks);
```
and loop through this array.